journals/apps/api/v1/views.py

```diff
@@ -62,3 +64,31 @@ def _revoke_access(self, order_number):
                     access_record.journal,
                     access_record.order_number)
         return HttpResponse()
+
+
+class UserPageVisitViewSet(viewsets.ModelViewSet):
+    """API for UserPageVisit model"""
+    lookup_field = 'user_id'
+    queryset = UserPageVisit.objects.all()
+    serializer_class = UserPageVisitSerializer
+    permission_classes = (IsAdminUser,)
+
+    def retrieve(self, request, *args, **kwargs):   # pylint: disable=unused-argument,useless-suppression
+        """
+        Returns the user's visited pages.
+        page_id is optional argument.
+        """
+        user_id = kwargs['user_id']
+        page_id = self.request.GET.get('page_id')
+        queryset = self.queryset.filter(user_id=user_id)
+        if page_id:
+            queryset = queryset.filter(page_id=page_id)
+        return Response(self.serializer_class(queryset, many=True).data)
+
+    @detail_route(url_path='lastvisit')
+    def user_last_visited_page(self, request, user_id):  # pylint: disable=unused-argument,useless-suppression
```

Comment:
not sure why we need this either? I propose we try to accomplish all the functionality by using querystring parameters on a single endpoint (i..e. /pagevisits/?lastvisited=true) or something like that

Reply:
although if this works fine I guess it's ok. I'm not that familiar with the @detail_route decorator

Reply:
@bill-filler  suppose we hit a URL .../pagevisits/lastvisit then that call will be routed to this function. And yes it is working fine.